pymc3/model.py

```diff
@@ -1518,3 +1472,343 @@ def all_continuous(vars):
         return False
     else:
         return True
+
+
+class ConstantNodeException(Exception):
+    pass
+
+
+def not_shared_or_constant_variable(x):
+    return (isinstance(x, theano.Variable) and
+            not(isinstance(x, theano.Constant) or
+                isinstance(x, theano.tensor.sharedvar.SharedVariable))
+            ) or (isinstance(x, (FreeRV, MultiObservedRV, TransformedRV)))
+
+
+_theano_autonamed_ops = [('.+\.T$',
```

Comment:
this could be moved into the function, right?
I'd also write the regular expressions turned as raw strings (`r''`), and check carefully that especially the first one does what you expect.